### ISSUE TITLE
Strange conditions in pipeflow initialization

### ISSUE THREAD

**Issue by dlohmeier** [e2nIEE/pandapipes#232]
I don't really understand some lines of code in the pipeflow procedure. If we look at the `pipeflow` function, we find [the following](https://github.com/e2nIEE/pandapipes/blob/1035df5decd81d2bcd6a2f1f4d34df3f93535e76/pandapipes/pipeflow.py#L68-L71): 

    create_lookups(net, NodeComponent, BranchComponent, BranchWInternalsComponent)
    node_pit, branch_pit = initialize_pit(net, Junction.table_name(),
                                          NodeComponent, NodeElementComponent,
                                          BranchComponent, BranchWInternalsComponent)

and in the pipeflow_setup these lines of code:

in [`create_lookups`](https://github.com/e2nIEE/pandapipes/blob/1035df5decd81d2bcd6a2f1f4d34df3f93535e76/pandapipes/pipeflow_setup.py#L414-L422):

    for comp in net['component_list']:
        if issubclass(comp, BranchComponent):
            branch_from, branch_table_nr = comp.create_branch_lookups(
                net, branch_ft_lookups, branch_table_lookups, branch_idx_lookups, branch_table_nr,
                branch_from)
        if issubclass(comp, NodeComponent) | issubclass(comp, BranchWInternalsComponent):
            node_from, node_table_nr = comp.create_node_lookups(
                net, node_ft_lookups, node_table_lookups, node_idx_lookups, node_from,
                node_table_nr, internal_nodes_lookup)

and in [`initialize_pit`](https://github.com/e2nIEE/pandapipes/blob/1035df5decd81d2bcd6a2f1f4d34df3f93535e76/pandapipes/pipeflow_setup.py#L337-L343):

    for comp in net['component_list']:
        if issubclass(comp, NodeComponent) | \
                issubclass(comp, BranchWInternalsComponent) | \
                issubclass(comp, NodeElementComponent):
            comp.create_pit_node_entries(net, pit["node"], node_name)
        if issubclass(comp, BranchComponent):
            comp.create_pit_branch_entries(net, pit["branch"], node_name)

As far as I can recall, the component classes should handle this kind of behaviour, as we want to be able to include new components without adapting the whole pipeflow code. So, if a component is a branch component and does not contain internal nodes, the `create_node_lookups` and `create_pit_node_entries` functions should simply do nothing. But that should be up to the component.

This code with global imports of different components also raises another issue that if inside a component we import something else that again references into pipeflow (in my case to the PipeFlowNotConvergedException that must be moved otherwise), we get a circular import. Of course that should be avoided.

**Comment by SimonRubenDrauz** [comment]
I included this because we discussed that for example in pure branch_models with zero_length. This component should not have create_pit_node_entries in it as this would be confusing. Another example would be pure node_components. In there it should not be possible to adapt for example create_pit_branch_entries. Therefore, I removed every function which should not be required in each component. This, however, required on the other side this if-condition. We could think about roll this change back.
